### PR TITLE
decoder: Add support for `Tuple` as `Constraint`

### PR DESCRIPTION
**Pull request by radeksimko**
Part of https://github.com/hashicorp/terraform-ls/issues/496

--- 

## TODO

 - [x] **completion**: `schema`: empty completion data
 - [x] **completion**: handling of multiline elements (+tests)
 - [x] **completion**: tests
 - [x] **hover**: tests
 - [x] **semantic tokens**: tests
 - [x]  **completion**: review existing tests
 - [x] **hover**: review existing tests
 - [x] **semantic tokens**: review existing tests
---

## Later
 - [ ] **reference targets**: Decide how to collect parent target for the whole tuple
 - [ ] **reference origins**: tests
 - [ ] **reference targets**: tests

### CODE REVIEW

File: decoder/expr_tuple_semtok_test.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+							},
+						},
+					},
+				},
+			},
+			`attr = [
+  keyword,
+  drowyek,
+  invalid,
+]`,
```
Comment (radeksimko) [inline decoder/expr_tuple_semtok_test.go:210]: Nitpick: This test may be more effective if the invalid element is in the middle or beginning, as it confirms that we don't return early and just skip that element.

File: decoder/expr_tuple_hover.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			return nil
+		}
+
+		if elemExpr.Range().ContainsPos(pos) {
+			expr := newExpression(tuple.pathCtx, elemExpr, tuple.cons.Elems[i])
+			return expr.HoverAtPos(ctx, pos)
+		}
+	}
+
+	content := fmt.Sprintf("_%s_", tuple.cons.FriendlyName())
```
[the remark this answers is not in the document]
Comment (dbanck) [reply decoder/expr_tuple_hover.go:29]: I'm leaning toward revisiting this when all expressions are merged and see if there are multiple locations where we can make use of `EmptyHoverData()`

**Comment by dbanck** [comment]
@radeksimko I implemented your suggestions. PTAL

**Review by radeksimko — COMMENTED** [review]
LGTM, thank you for making those changes.

**Review by dbanck — APPROVED** [review]

**Pull request merged**